app/services/fixity_checker.rb

```diff
@@ -0,0 +1,117 @@
+# FixityChecker.new(asset).check
+# FixityChecker.new(asset).prune_checks
+# This first version of the fixity checker only checks
+# for actual discrepancies between already-recorded
+# checksums and calculated ones.
+#
+# This means:
+# Assets with nil files are ignored.
+# Files with nil checksums are ignored.
+
+class FixityChecker
+  attr_reader :asset
+
+  NUMBER_OF_RECENT_PASSED_CHECKS_TO_KEEP = 5
+  CHECK_CYCLE_LENGTH = 7
+
+
+  # @param work [Work] Work object, it's members will be put into a zip
+  # @param callback [proc], proc taking keyword arguments progress_i: and progress_total:, can
+  #   be used to update a progress UI.
+  def initialize(asset)
+    raise ArgumentError.new("Please pass in an Asset.") unless asset.is_a? Asset
+    @asset = asset
+  end
+
+  #FixityChecker.new(asset).check
```

Comment:
Needs more docs saying what it will do (create a FixityCheck log record in db? And/or delete old ones?), and what the `sieve_integer` arg does.

I'm not sure about your approach to `sieving` -- although it is quite smart (nice job), I am not sure if we want a way to avoid even _fetching_ the things that we are not going to be checking. The "sifting"/"sieve" stuff needs to be better-documented if we're going to keep it, but let's talk about it first and see if we want to do it this way or not. 

if we keep doing it this way, I think the sifting logic should not be in `check` -- if you call `check`, you should get a check. The sifting stuff should be lifted to somehow be outside of `check`, to decide if you want to call `check`. 

But let's talk through the overall sifting approach before you spend time making any changes to it.